src/main/scala/is/hail/methods/VEP.scala

```diff
@@ -274,19 +320,75 @@ object VEP {
               printElement,
               _ => ())
 
-            val kt = if (csq)
-              jt.filter(s => !s.isEmpty && s(0) != '#')
-                .map { s =>
-                  val x = csq_regex.findFirstIn(s).get
-                  (variantFromInput(s), x.substring(4))
+            val original_variant = block.map({
+              v => (v.copy(altAlleles = v.altAlleles.filter(_.alt != "*")), v)
+            }).toMap
+
+            val kt = jt
+              .filter(s => !s.isEmpty && s(0) != '#')
+              .map { case s =>
+                if (csq) {
+                  val vvep = variantFromInput(s)
+                  if (!original_variant.contains(vvep))
+                    fatal(s"VEP output variant ${ vvep } not found in original variants.\nVEP output: $s")
+
+                  val v = original_variant(vvep)
+                  val x = csq_regex.findFirstIn(s)
+                  x match {
+                    case Some(value) =>
+                      val tr_aa = value.substring(4).split(",")
+                      if (vvep != v && allele_num_index > -1) {
+                        val alleleMap = getNonStarAlleleMap(v)
+                        (v, tr_aa.map({
+                          x =>
+                            val xsplit = x.split("\\|")
+                            val allele_num = xsplit(allele_num_index)
+                            if (allele_num.isEmpty)
+                              x
+                            else
+                              xsplit
+                                .updated(allele_num_index, alleleMap.getOrElse(xsplit(allele_num_index).toInt, -1))
+                                .mkString("|")
+                        }): IndexedSeq[Annotation])
+                      }
+                      else
+                        (v, tr_aa: IndexedSeq[Annotation])
```

Comment:
Thanks, great point! I actually already have a PR ready on top of this one to add the attribute stuff and that also annotates the VEP CSQ header in the attributes. This is quite important since it defines what that hellish pipe-delimited stuff means. 
But I agree, `force` can go :)